galaxy_skeleton2.R

```diff
@@ -31,29 +35,41 @@ raw_file <- readLines(con = "?")
 
 # What do you need to replace the two question marks with in order to extract
 # the first two letters?
-substr(x = raw_file, start = ?, stop = ?)
-
+Lines <- substr(x = raw_file, start = 1, stop = 2)
+Lines
 # The next step is then to find out *which* line starts with "--", and pick out
 # the first one. This can be done in a nice little pipe, where you have to fill
 # out the question marks and the missing function names:
-L <- 
-  (substr(x = raw_file, start = ?, stop = ?) == "?") %>% 
-  function_that_returns_the_index_of_all_TRUES %>% 
-  function_that_picks_out_the_minimum_value
+# L <- 
+#  (substr(x = raw_file, start = 1, stop = 3) == "--") %>% 
+#  function_that_returns_the_index_of_all_TRUES %>% 
+#  function_that_picks_out_the_minimum_value
+
+L <- (substr(x = raw_file, start = 1, stop = 2) == "--") %>%  # generates TRUE/FALSE list
+        which() %>% # print index of TRUES
+        min() # selects minimum index (the first position)
+  
+
 
 # Save the variable descriptions (i.e. the information in lines 1:(L-2)) in a
 # text-file for future reference using the cat()-function. The first argument is
 # the information that we want to print out. In order to get each element in the
 # "raw_file"-vector on a separate line we also provide the sep-argument, where
 # we put the "end-of-line"-character "\n". We also need to come up with a file
 # name. Replace the question marks:
-cat(?, sep = "\n", file = "?")
+# cat(?, sep = "\n", file = "?")
+
+cat(raw_file[1:(L-2)], sep = "\n", file = "descriptions")
```

Comment:
remember to add a postfix to the file name, such as ".txt" to specify the file type.